### ISSUE TITLE
AbrahamsonEtAl2014 fails when instantiated with valid.gsim()

### ISSUE THREAD

**Issue by cbworden** [gem/oq-engine#7363]
You suggested using openquake.hazardlib.valid.gsim() to turn a string into an instance of a GMM. This seems to work in general, but in some cases, it does not. I think I understand the problem, but first I'll illustrate it with AbrahamsonEtAl2014:
```
[igskcicgltgm016:coremods](master) <312> ipython
Python 3.8.12 (default, Oct 12 2021, 06:23:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from openquake.hazardlib.valid import gsim

In [2]: ask14 = gsim('AbrahamsonEtAl2014')

In [3]: ask14
Out[3]: [AbrahamsonEtAl2014]

In [4]: ask14.region
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-fa0f2550e26a> in <module>
----> 1 ask14.region

AttributeError: 'AbrahamsonEtAl2014' object has no attribute 'region'

In [5]: from openquake.hazardlib.imt import PGA

In [6]: ask14.compute(None, [PGA()], None, None, None, None)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-24cc0d4391dd> in <module>
----> 1 ask14.compute(None, [PGA()], None, None, None, None)

~/anaconda3/envs/shakemap/lib/python3.8/site-packages/openquake/hazardlib/gsim/abrahamson_2014.py in compute(self, ctx, imts, mean, sig, tau, phi)
    487             # compute median sa on rock (vs30=1180m/s). Used for site response
    488             # term calculation
--> 489             sa1180 = np.exp(_get_sa_at_1180(self.region, C, imt, ctx))
    490 
    491             # For debugging purposes

AttributeError: 'AbrahamsonEtAl2014' object has no attribute 'region'

```
As you can see, it looks like the "region" attribute has never been set. But it is clearly set in the `__init__()` method:
```
In [7]: ask14.__init__()

In [8]: ask14.region

In [9]: 
```
And everything is fine. So I think the problem is [here](https://github.com/gem/oq-engine/blob/fe2b790612a3c2f009502b42fe9ab442f15ed4b8/openquake/hazardlib/valid.py#L140):
```
        gs = object.__new__(gsim_class)
```
The object is instantiated with `__new__()`, but by some obscure rules of Python `__init__()` is never called. So I think all that is needed is for the following line to be:
```
        gs.__init__(**kwargs)
```
But there may be other (possibly more elegant) ways to achieve the same goal.

This is all using Openquake Engine 3.12, Python 3.8, Mac OSX 10.15.7.

**Comment by micheles** [comment]
Thank you for the analysis, the culprit is certainly the `__new__`. Notice that there is a `__new__` without in an `__init__` on purpose. The code in ``valid`` is many years old and at that time GMPEs had and optional ``init`` method that was called later on, so it was not called here to avoid performing a possibly expensive operation twice. Now that logic disappeared but the ``_new__`` remained. It should possibly to perform a regular instantiation now, I will do some test.

**Comment by micheles** [comment]
The fix has been backported to engine-3.12, just update the branch, or use the universal installer with ``--version=engine-3.12``

**Comment by cbworden** [comment]
Thank you for addressing this issue so quickly. I have installed OQ using the master repo, and everything is working smoothly now. This appears to be a good solution for us.

However, I did try try to install with 3.12 and it appeared to be the original code. I looked on github and 3.12 seemed not to have been updated. I double-checked by downloading the 3.12 assets, and they showed the original code, as well. Our typical install is with pip and looks like:
```
pip install --upgrade --no-dependencies https://github.com/gem/oq-engine/archive/refs/tags/v3.12.0.tar.gz
```
This is done in an environment with all of the necessary dependencies preinstalled. The install that worked was:
```
pip install --upgrade --no-dependencies git+https://github.com/gem/oq-engine.git
```
Which is just the master repo. Checking the code in each, I found the master had the new code and 3.12 had the original code. Maybe I just looked too soon and github hadn't updated 3.12 yet?


Thanks again for the fix.

**Comment by micheles** [comment]
This is why I wrote "please update the branch, or use the universal installer with --version=engine-3.12". If you are using pip, the right call to get the branch from github is something like "pip install https://github.com/gem/oq-engine/archive/engine-3.12.zip" (I did not try it, but it should be close). We did not make a new official release 3.12.2 since we had just released 3.12.1 and we prefer to wait a bit between releases to see if other bugs are reported. Anyway, you can use master too, since no changes to the GMPE API are scheduled for the foreseeable future and it should be stable again, now.
